sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/kafka/AzureKafkaConfigurationUtils.java

```diff
@@ -97,14 +98,26 @@ public static boolean needConfigureSaslOAuth(Map<String, Object> sourcePropertie
     private static boolean meetSaslOAuthConditions(Map<String, Object> sourceProperties) {
         String securityProtocol = (String) sourceProperties.get(SECURITY_PROTOCOL_CONFIG);
         String saslMechanism = (String) sourceProperties.get(SASL_MECHANISM);
-        if (securityProtocol == null || (SECURITY_PROTOCOL_CONFIG_SASL.equalsIgnoreCase(securityProtocol)
-                && (saslMechanism == null || SASL_MECHANISM_OAUTH.equalsIgnoreCase(saslMechanism)))) {
+        String jaasConfig = (String) sourceProperties.get(SASL_JAAS_CONFIG);
+        if (meetSecurityProtocolConditions(securityProtocol) && meetSaslMechanismConditions(saslMechanism)
+            && meetJaasConditions(jaasConfig)) {
             return true;
         }
         LOGGER.info(LOG_OAUTH_AUTOCONFIGURATION_RECOMMENDATION, saslMechanism);
         return false;
     }
 
+    private static boolean meetSecurityProtocolConditions(String securityProtocol) {
```

Comment:
Should this be `meetSaslConditions?

Reply:
I see, here the fuction should be named according to the expected conditions.